addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/discovery/DiscoveryListener.java

```diff
@@ -0,0 +1,112 @@
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.net.UnknownHostException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DiscoveryListener} is responsible for listening on the UDP socket for fan discovery messages.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+public class DiscoveryListener {
+    private final Logger logger = LoggerFactory.getLogger(DiscoveryListener.class);
+
+    private final String BCAST_ADDRESS = "255.255.255.255";
+    private final int SOCKET_RECEIVE_TIMEOUT = 300;
+
+    private final String POLL_MESSAGE = "<ALL;DEVICE;ID;GET>";
+
+    DatagramSocket dSocket;
+    DatagramPacket rcvPacket;
+    byte[] rcvBuffer;
+    InetAddress bcastAddress;
+    byte[] bcastBuffer;
+    DatagramPacket bcastPacket;
+
+    BigAssFanDevice device;
+
+    public DiscoveryListener() throws IOException, SocketException {
+        logger.debug("DiscoveryListener opening UDP broadcast socket");
+        dSocket = null;
+        device = new BigAssFanDevice();
+        try {
+            // Create a socket on the UDP port and get send & receive buffers
+            dSocket = new DatagramSocket(BAF_PORT);
+            dSocket.setSoTimeout(SOCKET_RECEIVE_TIMEOUT);
+            dSocket.setBroadcast(true);
+            rcvBuffer = new byte[256];
+            rcvPacket = new DatagramPacket(rcvBuffer, rcvBuffer.length);
+            bcastAddress = InetAddress.getByName(BCAST_ADDRESS);
+            bcastBuffer = POLL_MESSAGE.getBytes(CHARSET);
+            bcastPacket = new DatagramPacket(bcastBuffer, bcastBuffer.length, bcastAddress, BAF_PORT);
+        } catch (UnknownHostException uhe) {
+            logger.warn("UnknownHostException sending poll request for fans: {}", uhe.getMessage());
+        } catch (UnsupportedEncodingException e) {
+            logger.warn("Unable to convert buffer to string using {} charset", CHARSET);
+        } catch (Exception e) {
+            logger.warn("Unexpected exception getting UDP socket: {}", e.getMessage());
+        }
+    }
+
+    public BigAssFanDevice waitForMessage() throws IOException {
+        try {
+            // Wait to receive a packet
+            rcvPacket.setLength(rcvBuffer.length);
+            dSocket.receive(rcvPacket);
+
+            // Process the received packet
+            device.reset();
+            device.setIpAddress(rcvPacket.getAddress().getHostAddress());
+            String message = (new String(rcvBuffer, 0, rcvPacket.getLength()));
+            device.setDiscoveryMessage(message);
+            logger.debug("RECEIVED packet of length {} from {}: {}", message.length(), device.getIpAddress(), message);
+
+        } catch (SocketTimeoutException ste) {
+            // Timed out on UDP socket read
+            return null;
```

Comment:
Because you return null here, another method has to handle that null, if you would not handle the SocketTimeoutException you don't need to return null here; you don't ignore (I see no logging etc) the error; and you don't need to check for null in the methods processing the response of this message. Example:

```java
+    private void processMessage(BigAssFanDevice device) {
 +        if (device == null) {
 +            return;
 +        }
```